src/main/java/co/cask/hydrator/plugin/DynamicPartitionFileSetParquetSink.java

```diff
@@ -105,7 +101,13 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     Partitioning.Builder partitionBuilder = Partitioning.builder();
     String[] partitionFields = config.fieldNames.split(",");
     for (int i = 0; i < partitionFields.length; i++) {
-      partitionBuilder.addStringField(partitionFields[i]);
+      // throw exception if the field used to partition does not exist
+      if(parsedSchema.getField(partitionFields[i]) == null) {
+        throw new IllegalArgumentException(String.format("Field %s is not a member of the input schema.",
```

Comment:
The wording here is strange. "Not a member"? Maybe "Field %s is not present in the input schema..."